lib/inputs/index.ts

```diff
@@ -20,3 +25,27 @@ export function getFileContent(
 
   return foundAppFiles;
 }
+
+export function getElfFileContent(
+  extractedLayers: ExtractedLayers,
+  searchedAction: string,
+): FilePathToElfContent {
+  const foundAppFiles = {};
+
+  for (const filePath of Object.keys(extractedLayers)) {
+    for (const actionName of Object.keys(extractedLayers[filePath])) {
+      if (actionName !== searchedAction) {
+        continue;
+      }
+
+      const elfFile = extractedLayers[filePath][actionName] as Elf;
+
+      if (!(elfFile.body && elfFile.body.programs && elfFile.body.sections)) {
+        throw new Error("elf file expected to contain programs and sections");
+      }
+      foundAppFiles[filePath] = extractedLayers[filePath][actionName];
```

Comment:
This overwrites `foundAppFiles[filePath]` with the last valid `extractedLayers[filePath][actionName]` right? Is it possible there are multiple valid actionNames for each filePath, or will there only ever be one?

Reply:
The combination of `file path` and `action` is always unique. And this function extracts specifically based on the action name of all `filePaths: actionResult` combination. So it's not possible to be overwritten. 
This is how we call the function https://github.com/snyk/snyk-docker-plugin/pull/308/files#diff-ef7473bc4617398b357238645eb8280929cc7617ebf63ad15e8fb1ea8d82cf34R154